pkg/credentials/credentials.go

```diff
@@ -266,11 +272,50 @@ func defaultValidationFunc(creds map[string]string) error {
 	return nil
 }
 
+func (f *fetcher) parseOpenstackCloudCredentials() error {
+	cloudName, ok := os.LookupEnv("OS_CLOUD")
+
+	if !ok {
+		return nil
+	}
```

Comment:
`os.LookupEnv` returns true even if it's set but is empty. I'd consider `clouds.yaml` only if there's some content in the environment variable, e.g. name of the configuration to be used.